src/backend/parser/gram.y

```diff
@@ -3850,7 +3850,7 @@ alter_table_cmd:
 					if (witham) 
 					{
 						if (strlen(witham) != strlen(n->name) || 
-							strncmp(n->name, witham, strlen(n->name) != 0))
+							strcmp(n->name, witham) != 0)
```

Comment:
Shouldn't it be
```c
strncmp(n->name, witham, strlen(n->name))
```

Reply:
no need , already test `strlen(witham) != strlen(n->name)`